docs/install/ha_mode.md

```diff
@@ -0,0 +1,40 @@
+# HA setup
```

Comment:
I don't understand how it works at the end. Could you please describe in some more detail how it works if you have two clusters with HA, especially which URL should be set for `root_url` and `ha.root_url` and what happens if one of the clusters is not available anymore.

Reply:
We have two clusters with HA as belows:
cluster A: 
 - root_url: a.infrabox.com
 - ha.root_url: infrabox.com
 - components: dashboard, scheduler, controller, api, checker, ...
 - cluster_name: a

cluster B:
 - root_url: b.infrabox.com
 - ha.root_url: infrabox.com
 - components: dashboard, scheduler, controller, api, checker, ...
 - cluster_name: b

I setup ingress for both root_url and ha.root_url, so every cluster can be accessed by it's root_url and ha.root_url.

The ha.root_url points to active and enabled cluster through LoadBalancer or DNS failover.
The url a.infrabox.com/api/status return HTTP 200 if it's active and enabled or HTTP 500.
The health checker behand DNS or LoadBalancer can monitor cluster status by monitor this api and switch automatically when the primary cluster is not healthy.

I make sure that the scheduler works well with multi scheduler by adding row level locks for jobs to be scheduled. One job will not be scheduled repeatedly. Jobs will only be scheduled to active and enabled clusters.

For github job. The webhook is pointed to ha.root_url, so GitHub will post changes to current primary cluster. There will not be repeated jobs.

For gerrit jobs. The trigger creates new build in a transaction. The build table has a unique constraint on `(project_id, build_number, restart_counter)`, so the build will be triggered only once. For gerrit-review, There is only one leader can perform review.

There is a checker in every cluster. The checker checks these items:
- pods status in namespace infrabos-system
- dashboard connection
- api connection
- storage upload/download test
We can add more tests in future.

If the checker fails for more than 3 times, it will mark the cluster inactive.
If the checker succeeds for more than 3 times, it will mark the cluster active.

The checker refreshes it's last_update_time repeatedly. A cluster will be marked inactive by others if it doesn't refresh last_update_time for 1 minute.